wandb/run_manager.py

```diff
@@ -1080,11 +1078,12 @@ def shutdown(self, exitcode=0):
         if self._run_status_checker:
             self._run_status_checker.shutdown()
 
+        self._run.history.close()
```

Comment:
Just for my understanding, can you explain why closing history earlier causes us to not lose data?

Reply:
Closing history flushes any remaining updates to disk and closes the file.  The file_pusher listens for changes to files on disk and syncs them to the server.  This code is awful, so there are likely other races, but this should be an improvment.